File: GA/ga_toolbox/population.m
% Generate N random solutions within the boundary
%
% input:
%    N: number of random solutions to generate
%    nParams: number of parameters
%    LR: lower boundary [nParams x 1]
%    UR: upper boundary [nParams x 1]
% output:
%    P: [N x nParams] random solutions
%
function P = population(N,nParams,LR,UR)

    P = zeros(N,nParams);
    for i = 1:nParams
        P(:,i) = (UR(i)-LR(i))*rand(N,1)+LR(i);
    end

end

File: GA/ga_toolbox/population.py
import numpy as np


def population(N, nParams, LR, UR):
    """
    Generate N random solutions uniformly within the boundary.

    Parameters
    ----------
    N       : int        number of random solutions to generate
    nParams : int        number of parameters
    LR      : array-like [nParams,]  lower boundary
    UR      : array-like [nParams,]  upper boundary

    Returns
    -------
    P : np.ndarray  [N x nParams]  random solutions
    """
    LR = np.atleast_1d(LR)
    UR = np.atleast_1d(UR)

    P = np.zeros((N, nParams))
    for i in range(nParams):
        P[:, i] = (UR[i] - LR[i]) * np.random.rand(N) + LR[i]

    return P